test/unit/throttle.spec.js

```diff
@@ -0,0 +1,99 @@
+import chai from 'chai';
+import throttle from '../../lib/throttle';
+
+class Editor {
+  counter = 0;
+
+  @throttle(500)
+  updateCounter1() {
+    this.counter++;
+  }
+
+  @throttle(500, false)
+  updateCounter2() {
+    this.counter++;
+  }
+
+  @throttle(500, false, false)
+  updateCounter3() {
+    this.counter++;
+  }
+}
+
+describe('@throttle', function () {
+  let editor;
+
+  beforeEach(function () {
+    editor = new Editor();
+  });
+
+  it('invokes function only once', function (done) {
+    editor.updateCounter1();
+    editor.counter.should.equal(1);
+
+    setTimeout(() => {
+      editor.counter.should.equal(1);
+      done();
+    }, 600);
+  });
+
+  it('invokes function only twice', function (done) {
+    editor.updateCounter1();
+    editor.updateCounter1();
+    editor.updateCounter1();
+    editor.updateCounter1();
+    editor.counter.should.equal(1);
+
+    setTimeout(() => {
+      editor.counter.should.equal(2);
+      done();
+    }, 600);
+  });
+
+  it('invokes function delay and only once if "leading" option is false', function (done) {
+    editor.updateCounter2();
+    editor.counter.should.equal(0);
+
+    // should still be 1 because 500ms hasn't yet passed
+    setTimeout(() => {
+      editor.counter.should.equal(0);
+    }, 400);
+
+    setTimeout(() => {
+      editor.counter.should.equal(1);
+      done();
+    }, 600);
+  });
+
+  it('not invokes function if "leading" and "trailing" options are both false', function (done) {
+    editor.updateCounter3();
+    editor.counter.should.equal(0);
+
+    // should still be 0 because leading call a canceled
+    setTimeout(() => {
+      editor.counter.should.equal(0);
+      editor.updateCounter3();
+    }, 400);
+
+    // should still be 0 because trailing call a canceled
+    setTimeout(() => {
+      editor.counter.should.equal(0);
+      done();
+    }, 600);
+  });
+
+  it('separate instances do not share timers', function (done) {
```

Comment:
test titles need to be a sentence structured to make sense with "it" at the beginning (the name of the function).

e.g.

```js
it('does not share timers between instances', function (done) {
```

Same issue above,

```js
it('does not invoke function as "leading" or "trailing" if options are false', function (done) {
```